test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/ObjectResultTests.cs

```diff
@@ -468,10 +475,148 @@ public async Task
+        }
+
+        [Theory]
+        [InlineData("text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+            "application/xml;charset=utf-8")] //Chrome
+        [InlineData("text/html, application/xhtml+xml, */*",
+            "application/json;charset=utf-8")] //IE
+        [InlineData("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "application/xml;charset=utf-8")] //Firefox
+        [InlineData("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "application/xml;charset=utf-8")] //Safari
+        [InlineData("text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
+            "application/xml;charset=utf-8")] //Opera
+        [InlineData("*/*",
+            "application/json;charset=utf-8")]
+        [InlineData("text/html,*/*;q=0.8,application/xml;q=0.9",
+            "application/xml;charset=utf-8")]
+        public async Task ObjectResult_PerformsContentNegotiation_OnAllMediaRangeAcceptHeaderMediaType(
+            string acceptHeader,
+            string expectedResponseContentType)
+        {
+            // Arrange
+            var objectResult = new ObjectResult(new Person() { Name = "John" });
+            var outputFormatters = new IOutputFormatter[] {
+                new HttpNoContentOutputFormatter(),
+                new TextPlainFormatter(),
+                new JsonOutputFormatter(),
+                new XmlDataContractSerializerOutputFormatter(XmlSerializerOutputFormatter.GetDefaultXmlWriterSettings())
+            };
+            var response = GetMockHttpResponse();
+
+            var actionContext = CreateMockActionContext(
+                                    outputFormatters,
+                                    response.Object,
+                                    requestAcceptHeader: acceptHeader,
+                                    respectBrowserAcceptHeader: true);
+
+            // Act
+            await objectResult.ExecuteResultAsync(actionContext);
+
+            // Assert
+            response.VerifySet(resp => resp.ContentType = expectedResponseContentType);
+        }
+
+        [Theory]
+        [InlineData("application/xml;q=0.9,text/plain;q=0.5", "application/xml;charset=utf-8", false)]
+        [InlineData("application/xml;q=0.9,*/*;q=0.5", "application/json;charset=utf-8", false)]
+        [InlineData("application/xml;q=0.9,text/plain;q=0.5", "application/xml;charset=utf-8", true)]
+        [InlineData("application/xml;q=0.9,*/*;q=0.5", "application/xml;charset=utf-8", true)]
+        public async Task ObjectResult_WildcardAcceptMediaType_AndProducesAttribute(
+            string acceptHeader,
+            string expectedResponseContentType,
+            bool respectBrowserAcceptHeader)
+        {
+            // Arrange
+            var objectResult = new ObjectResult(new Person() { Name = "John" });
+            objectResult.ContentTypes.Add(MediaTypeHeaderValue.Parse("application/xml"));
```

Comment:
I do not see the produces attribute being used directly here.  ( I know it will result in the same thing),  I suggest  use produces in a functional test to cover this scenario and change the test name  to reflect what it actually tests.